docs/docsite/rst/dev_guide/developing_modules_documenting.rst

```diff
@@ -29,30 +30,93 @@ Include it in your module file like this:
+  * A longer description than `short_description`, which goes into more detail, generally two or more sentences.
+  * Must be written in full sentences, i.e. with capital letters and fullstops.
+  * Shouldn't mention the name module.
+* ``version_added:``
+  The version of Ansible when the module was added.
+  This is a `string`, and not a float, i.e. ``version_added: "2.1"``
+* ``author:``
+  In the form ``First Last (@GitHubID)``, use a multi-line list if there is more than one.
+* ``options:``
+  One per option name
+
+  * ``description:``
+
+    * Detailed explanation of what this option does and what it does. It should be written in full sentences.
+    * Should not list the option values (that's what ``choices:`` is for, though it should explain `what` the values do if they aren't obvious.
+    * If an argument takes both True/False and Yes/No, the documentation should use True and False.
+    * If an optional parameter is sometimes required this need to be reflected in the documentation, e.g. "Required when I(state=present)."
+    * Mutually exclusive options must be documented as the final sentence on each of the options.
+  * ``required:``
+    Only needed if true, otherwise it is assumed to be false.
+  * ``default:``
+    * If `required` is false/missing, `default` may be specified (assumed 'null' if missing).
+    * Ensure that the default parameter in docs matches default parameter in code. The default option must not be listed as part of the description.
+  * ```choices``
+    List of option values. Should be absent if empty.
+  * ``aliases:``
+    List of option name aliases, generally not needed. Should be absent if empty.
+  * ``version_added:``
+    Only needed if this option was extended after initial Ansible release, i.e. this is greater than the top level `version_added` field.
+    This is a string, and not a float, i.e. `version_added: "2.3"`
+* ``requirements:``
+    List of requirements, and minimum versions (if applicable). These are the requirements on the host that executes the module.
+* ``notes:``
+    Details of any important information that doesn't fit in one of the above sections, for example if ``check_mode`` isn't supported, or a link to external documentation.
+
+.. note:: Versions should be strings
+
+    In general, care should be taken to list versions as strings rather than floats.  This has
+    several reasons:
+    * Versions may not always be floats, for instance "2.1.0" or "2.0alpha1".
+    * Versions may not compare correctly if used as a float.  A float will tempt
+       naive comparison between 2.3 and 2.12 which won't generate the correct
+       ordering.  Strings "2.3" and "2.12" require a bit of code which will let the
+       programmer do it correctly.
+
+
+EXAMPLES block
+''''''''''''''
+
+The EXAMPLES section, just like the documentation section, it is required for all new modules.
+
+Examples should demonstrate real world usage, and be written in multi-line YAML format in plain text in.
```

Comment:
Examples should demonstrate real world usage, and be written in multi-line plain-text YAML format.